### ISSUE TITLE
TODO ideas/requests from TG/4pda/folksinperson

### ISSUE THREAD

**Issue by nikitasius** [nikitasius/Telegraher#17]
* 1 [ ] improve the voice message quality (cause it sucks compared to iphone)
* 2 [ ] add the ghost mode per user and global
* 3 [ ] personal local moderarion (delete/edit messages from the other folks locally + remove "deleted" flag) 
* 4 [ ] wipe all own messages in any group chat
* 5 [ ] give possibility to remove "ALL" folder + add some private cases (i.e. "private messages", etc)
* 6 [ ] support system AND vanilla fonts
* 7 [ ] add support for external fonts (any)
* 8 [ ] change text size for the UI elements (labels, rounds etc)
* 9 [ ] option to disable reactions in UI
* 10 [ ] option to disable strickers by type (static, animated, video)
* 11 [ ] option to disable gifs
* 12 [ ] change stickers scale in %
* 13 [ ] hide TG accounts in system
* 14 [ ] add db encryption for TG
* 15 [ ] add autoupdate from github
* 16 [ ] add multiple translators
* 17 [ ] fix map issue
* 18 [ ] add map providers
* 19 [ ] add settings import/export
* 20 [ ] add database import/export into sql/binary
* 21 [ ] add chat export into csv/json
* 22 [ ] remove account limit 6 to ~
* 23 [ ] disable sensors in the app when app in a background
* 24 [ ] add icons to folders
* 25 [ ] remove blur from thumbnails
* 26 [ ] fix media w/ a timer in classic cloud chats (yezzz)

**Comment by alvkN** [comment]
you can use implementation of 5 from [CatoX](https://github.com/CatogramX/CatogramX), it better than neko and others (swipe back doesn't open hidden ALL tab)

**Comment by nikitasius** [comment]
> you can use implementation of 5 from [CatoX](https://github.com/CatogramX/CatogramX), it better than neko and others (swipe back doesn't open hidden ALL tab)

Thanks i will check for this :+1:

**Comment by OshekharO** [comment]
- [ ]  Forward Without Quoting ( in Channels & Groups )
- [ ] Delete From Cache ( in Channels & Groups )
- [ ] Mark all read
- [ ] Save to cloud ( in Channels & Groups )
- [ ] Proxy from nekogram
- [ ] Copyright Infringement Blocked Bots, channels & groups Unlocked
- [ ] Option to delete all my messages from the group
- [ ] Add "Go to first message" to three dot menu ( in Channels & Groups )
- [ ] Inbuilt Updater ( Like Nekogram or mdgram )
- [ ] Add Amoled Theme As default
- [ ] Unlimited Pinned option
- [ ] Custom audio bitrate in group Voip (64gram)
- [ ] [Redesign] Settings like mdgram
- [ ] More optimization ( overall )
- [ ] Add to favourites to three dot menu
- [ ] Voice Changer 
- [ ] Manage inactive channels ( hulugram )
- [ ] Copy selection ( in Channels & Groups )
- [ ] Add Independent Section For Bots List ( botlist.net )

@nikitasius check this out

**Comment by b3pio** [comment]
@nikitasius Any chance to publish on F-droid / IzzyOnDroid repo? Also can you consider removing Google.Firebase?

**Comment by nikitasius** [comment]
@OshekharO cool :+1: 

@b3pio it's a good idea. Actually TG want us to stop. I sent them some questions and i need their answers. When they will answer me i will tell you better. Their answer will be shared with the chat (TG and Matrix).
For me safety of users is the most important but some features can show different app behavior and signal "hey, it;s probably telegraher". So, if there are no negociations with TG some **future** features cannot be implemented because they will expose us. But if there are negociations i will tell in chat simply "what do you prefer folks, have feature A and B or have app in fdroid?" sure if nothing from fdroid can ban us.

**Comment by nikitasius** [comment]
Well, i spoken with a Barinsta dev and looks like fdroid will not remove us if TG will ask them. Sounds good.
I will prepare the app for fdroid. Hope it they will be able to build is fine.

But remember:
* if fdroid build it, you/me/anyone will need to remove old app due another signature (they will sign with their fdroid keys)

**Comment by b3pio** [comment]
@nikitasius this is great news! you may need to remove this from code for f-droid approval.

Google Firebase Analytics

*Google Firebase Analytics
2com.google.firebase.analytics.

**Comment by nikitasius** [comment]
> @nikitasius this is great news! you may need to remove this from code for f-droid approval.
> 
> Google Firebase Analytics
> 
> *Google Firebase Analytics 2com.google.firebase.analytics.

Hi, yep. I wanna clean it a bit.

**Comment by nikitasius** [comment]
@OshekharO well about your list:

- [ ]  ~~Forward Without Quoting ( in Channels & Groups )~~  vanilla have it in 2 clicks
- [ ] ~~Delete From Cache ( in Channels & Groups )~~  vanilla have it in the menu when you select the chat
- [ ] ~~Mark all read~~ debug menu have it "read all chats"
- [ ] Save to cloud ( in Channels & Groups ) - INFO - please explain
- [ ] Proxy from nekogram - INFO - which ones?
- [ ] ~~Copyright Infringement Blocked Bots, channels & groups Unlocked~~ - TG have GEO bans and PLATFORM bans. GEO ban can be bypassed using another sim. Platform bans can be bypassed using another platform (all, android, ios and ms/ws for windows). Actually all works on debian. I can "emulate a linux", but to do it better need a lot of changed in a code. So faster to install a debian desktop client.
- [x] Option to delete all my messages from the group - number 4
- [x] Add "Go to first message" to three dot menu ( in Channels & Groups ) - added as 35, good idea
- [ ] ~~Inbuilt Updater ( Like Nekogram or mdgram )~~ - app will be updated by fdroid.
- [ ] ~~Add Amoled Theme As default~~ i'm on IPS :stuck_out_tongue:
- [x] Unlimited Pinned option - added as 36, it will be client side
- [x] Custom audio bitrate in group Voip (64gram) - added as 37, good
- [ ] ~~[Redesign] Settings like mdgram~~ i dont know mdgram. Own settings are on special page. Settings which extends basic functionality where the basic functionality.
- [ ] ~~More optimization ( overall )~~ means nothing :)
- [ ] Add to favourites to three dot menu - INFO, what do you mean?
- [x] Voice Changer  - already as 33
- [ ] ~~Manage inactive channels ( hulugram )~~ channels with no posts? well just scroll and leave.
- [ ] Copy selection ( in Channels & Groups ) - INFO, need details
- [ ] ~~Add Independent Section For Bots List ( ime Messenger {botlist.net or botsarchive.com} )~~ - in vanilla you can filter the bots, but if you wanna add 3rd party API - no. It's data share&collection + dependancy from 3rd party service. Everyone can open a website, search and add a bot.
- [x] Keep the original file name when downloading files - added as 38 dnload and 39 for sending. Via checkbox, default is disabled.
- [x] Option to ignore Android-only content restrictions - already ignore it and the restriction array. if you mean "Button to enable content-sensitive" it;s addded as 40
- [ ] ~~Retract Vote ( mdgram & plus messenger )~~ - vanilla offer it

**Comment by AlbertEienstein** [comment]
@nikitasius just make your own f-droid repo...

**Comment by b3pio** [comment]
@AlbertEienstein, @IzzySoft 's repo would probably be a better option than maintaining his own repo for less work and more exposure for users to find this app. So either Izzys can be the repo it lives on or just as a stepping stone to main f-droid repo.

**Comment by AlbertEienstein** [comment]
> @AlbertEienstein, @IzzySoft 's repo would probably be a better option than maintaining his own repo for less work and more exposure for users to find this app. So either Izzys can be the repo it lives on or just as a stepping stone to main f-droid repo.

or Just A in-app updater so app can be downloaded via github releases and be updated automatically?

**Comment by b3pio** [comment]
> > @AlbertEienstein, @IzzySoft 's repo would probably be a better option than maintaining his own repo for less work and more exposure for users to find this app. So either Izzys can be the repo it lives on or just as a stepping stone to main f-droid repo.
> 
> or Just A in-app updater so app can be downloaded via github releases and be updated automatically?

If its published on Izzys repo then it keeps same dev keys as github release so it will then work either way you install. Its not signed with different keys like it would be in f-droid main repo.

**Comment by IzzySoft** [comment]
All true what @b3pio writes. But keep in mind that the per-app size limit in my repo is 30M – even the smallest current APK is already beyond that. As apps tend to rather grow than shrink, I'm hesitant adding an app that's already over this limit to start with.

**Comment by nikitasius** [comment]
Hi folks, thanks for a good ideas here :+1: 
@AlbertEienstein it;s interresting but while there are many other passionate folks who do it better than me, it's great.
@IzzySoft repo sounds good. @b3pio i will try to clean this fork (remove completely fcm) and to see what will be changed and how it will affect the final size.
If i will use my own keys (actually this repo use the default keys, everyone knows them), i will generate them and i will add into github secrets (if i use CI/CD here). Actually i did not changed it because idk if it goes to fdroid or any another repo (and another repos can build and sign with their own keys).

**Comment by IzzySoft** [comment]
@nikitasius seeing your enthusiasm, I certainly don't want to be the one on the brakes here. 30M is no hard limit (it's because all this runs on my personal resources only and I somehow have to limit it); 32M would be a technical limit (for uploading the file to VirusTotal via their APIv2 which I use, to have it scanned), but as long as exceptions are few I can work around that (using some v3 calls I've implemented). So those 2M are mostly a safety buffer.

TL;DR: if you can get the size down a bit that would be great, but if it's a little (!) over 30M I can close an eye or two on that. Let me know then, and I'll take a look if the other conditions are met – when you're ready.

Concerning FCM: May I feed your enthusiasm by pointing you to some [alternative push services](https://gitlab.com/-/snippets/1896503#push) like @UnifiedPush – or if needed, also to full Firebase alternatives like [appwrite](https://github.com/appwrite/appwrite) or [Supabase](https://supabase.com/)? :smiley:

**Comment by nikitasius** [comment]
> Concerning FCM

Hi, @IzzySoft recently i removed the FCM :)
picked some useful stuff from the foss one, include osmand maps.

**Comment by IzzySoft** [comment]
Hi @nikitasius – thanks for the heads-up! I just took a quick look: we're getting close. It's slightly beyond the size limit, but as pointed out I can close an eye on that. I just picked the armeabi build, but my library scanner still complains:

```
Offending libs:
---------------
* Android Vision (/com/google/android/gms/vision): NonFreeDep
* Google Mobile Services (/com/google/android/gms): NonFreeDep
* Firebase (/com/google/firebase): NonFreeNet,NonFreeDep
* ML Kit (/com/google/mlkit): NonFreeDep,Tracking

4 offenders.
```

Vision is probably pulled in by ML Kit. So are Firebase and GMS. All of them are proprietary/non-free. I'm not sure what you use them for, so I cannot name potential replacements. But `Tracking` with a personal/private messenger is not the best idea I'd say, so it would be good to get rid of ML Kit.

For the listing I'd also need some description and screenshots, ideally provided via [Fastlane structures](https://gitlab.com/-/snippets/1895688) in the app's repo here (so you can easily adjust them whenever you feel the need, and my updater can take care to pull that automatically with the next release).

To make your donation links easier to discover, you could use a `.github/FUNDING.yml` (which my updater then would take care for as well). That's a Github standard, and yours would look like this:

```yaml
custom: ['https://yoomoney.ru/to/410015481871381','https://paypal.me/nikitasius'] # up to 4 custom URLs
```

They seem to have no field for BTC or other crypto, and I currently cannot find the documentation. Other available fields would include:

```yaml
github: # Replace with up to 4 GitHub Sponsors-enabled usernames e.g., [user1, user2]
patreon: # Replace with a single Patreon username
open_collective: # Replace with a single Open Collective username
ko_fi: # Replace with a single Ko-fi username
tidelift: # Replace with a single Tidelift platform-name/package-name e.g., npm/babel
community_bridge: # Replace with a single Community Bridge project-name e.g., cloud-foundry
liberapay: # Replace with your Liberapay user name
issuehunt: # Replace with a single IssueHunt username
otechie: # Replace with a single Otechie username
```

**Comment by heller390** [comment]
> 

@nikitasius Is there the option to disable content filtering ?! I am unable to find it out.

**Comment by nikitasius** [comment]
@heller390 you mean same like desktop have? i did not added it nor checked how does it work.
By default we ignore all what we can ignore. But i will take a look to this option.

Actually you can use tg desktop on linux to disable it.

**Comment by TeaqariaWTF** [comment]
Can translation feature be added from owlGram?  The Owlgram translation feature is very good. There is a vibration that occurs when switching between channels by swiping.

**Comment by TeaqariaWTF** [comment]
![Screenshot_20220604-072341_OwlGram](https://user-images.githubusercontent.com/90970076/171982651-04f8ea4d-4cdd-47a2-863e-940e699fb003.png)
While saving a file to the device, we wait for the download to complete, just like this. Nekox also has a download manager, we don't have to wait on this screen while downloading

**Comment by TearsOfTheEarth** [comment]
Hi,

Cheers for the quick reaction removal.
A few suggestions if I may:
- [ ] Option to completely hide large "unmute" button at the bottom that takes a lot of valueable space and only good for misclicks (option is still available under dots menu)
- [ ] Option to hide (collapse to a single line all other unread occurencoes maybe with a reference to the first seen) similar messages that will save space and time seeing forwards and annoying ads that are crossposted to multiple channels. Though this might be challenging and resource consuming to implement as it probably requires to run through all unread messages and do full mesh comparison unless I miss a better algorithm. As an interim forwards from channel you are subscribed to might be collapsed this way.

**Comment by nikitasius** [comment]
Hi, @IzzySoft , if you're still intrerresting, i will be glad to integrate with your repo. Normally i removed non-foss trash from client.
Actual betas aren't on github's CI/CD (cause if i will share them via github folks will think that it's release, even if i mentioned beta...), so here are the APKs, built on a server, you can analyze them: https://git.evildayz.com/Telegraher/betas/

**The release will be in github releases via github actions.**

Also some handy changes:
* i wanna change readme/description, make it shorter (keeping legacy big text somewhere in a repo)
* there are 4 apks, not 8. I removed sdk23 cause after cleaning permissions there are no differ now
* new 8.8.x version will be incompatible with the oldones, so it will require a clean install
* i will check if github will allow me to build all 4 apks in 1 release (before it was an issue due CI/CD limitations here)

Also, because new version will not work as an update for the old ones i wanna change app signature. I will make my own certificates for this. The purpose:
* if someone will mod the app they can't install it over the actual app as they can do it now (the situation when someone got your phone, it's non-rooted and they simply install modded version to extract your data the easy way)

Also why i find good to use your repo:
* apps are built on github or no matter which CI/CD where project are linked
* folks can install apks from your repo via fdroid app or directly from the github

**Comment by nikitasius** [comment]
![image](https://user-images.githubusercontent.com/3670331/183378041-9c57bfb6-cb1b-47f4-a305-653836d01127.png)
Also it could be a bundle, just 1 apk with all architectures.
apk size is x2, but installed size +60Mb more.

**Comment by IzzySoft** [comment]
> so here are the APKs, built on a server, you can analyze them

My APK library scanner confirms there seem to be no offending libs. But unfortunately the APKs are 35M+ – which is beyond the limit of 30M per application in my repo. This might be due to the beta having some additional debug code in, I cannot tell.

> i wanna change readme/description

ideally, set up [Fastlane structures](https://gitlab.com/-/snippets/1895688) with the app's repo, so you're in control of how the app is presented. For formatting of the full description, you can find [some hints here](https://gitlab.com/IzzyOnDroid/repo/-/wikis/FAQ).

> there are 4 apks

I can only pick one per package name.

> because new version will not work as an update for the old ones i wanna change app signature.

In that case, consider changing package name (`applicationId`) instead/too to avoid confusion.

> why i find good to use your repo

Thanks – and yes, that was part of the idea: keeping the barriers low. But at the same time, keeping the door to f-droid.org open and focused, if possible :wink:

**Comment by MasterKia** [comment]
- [ ] Option to hide third-party promotions or sponsorships in channels (advertisements by the channel owner).

How to detect if a message is advertisement?
If the message contains links or usernames to other Telegram channels then it's possibly an advertisement.
